Sources/COWMacros/COWExpansionFactory.swift

```diff
@@ -218,30 +223,257 @@ internal class COWExpansionFactory<Context: MacroExpansionContext> {
+    // By resorting to an underscored attribute we can correct this with
+    // minimal effort.
+    let dynamicDispatchFixup: DeclSyntax = """
+    @_implements(Swift.Equatable, ==(_:_:))
+    public static func _$equals(lhs: Self, rhs: Self) -> Bool {
+      return lhs == rhs
+    }
+    """
+    associatedMembers.append(dynamicDispatchFixup)
+  }
+  
+  private func forwardCodingKeysForDerivedStorageIfNeeded(
+    members: inout [MemberBlockItemSyntax],
+    protocols: inout [InheritedTypeSyntax],
+    associatedMembers: inout [any DeclSyntaxProtocol]
+  ) {
+    // Bail out early if `appliedStructDecl` does not declare CodingKeys.
+    guard appliedStructDecl.memberBlock.members.contains(where: {
+      if let enumDecl = $0.decl.as(EnumDeclSyntax.self),
+         enumDecl.name.trimmed.text == "CodingKeys",
+         let inheritedTypes = enumDecl.inheritanceClause?.inheritedTypes,
+         inheritedTypes.containsType(named: "CodingKey") {
+        return true
+      }
+      // We have no way to check the aliased type, so just make a guess (that
+      // it is indeed a CodingKeys enum).
+      if let typealiasDecl = $0.decl.as(TypeAliasDeclSyntax.self),
+         typealiasDecl.name.trimmed.text == "CodingKeys" {
+        return true
+      }
+      return false
+    }) else {
+      return
+    }
+    
+    // Bail out early if we are not conforming to anything Codable related.
+    let conformingToCodable = protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return codableProtocolNames.contains(name)
+    })
+    let conformingToEncodable = conformingToCodable || protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return encodableProtocolNames.contains(name)
+    })
+    let conformingToDecodable = conformingToCodable || protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return decodableProtocolNames.contains(name)
+    })
+    guard conformingToCodable ||
+          conformingToEncodable ||
+          conformingToDecodable else {
+      return
+    }
+    
+    // Make a typealias in the derived storage pointing to the CodingKeys in
+    // `appliedStructDecl` which allow the compiler to synthesize the actual
+    // coding functions.
+    let typealiasName: TokenSyntax = "CodingKeys"
+    members.append(
+      MemberBlockItemSyntax(
+        decl: TypeAliasDeclSyntax(
+          name: typealiasName,
+          initializer: .init(
+            value: MemberTypeSyntax(
+              baseType: IdentifierTypeSyntax(name: appliedStructDecl.name),
+              name: typealiasName
+            )
+          )
+        )
+      )
+    )
+    
+    // Expand coding functions in `appliedStructDecl`, forwarding the
+    // invocation to the derived storage.
+    let storageName = appliedStructDecl.copyOnWriteStorageName
```

Comment:
The `storageName` (which is the name of the storage instance in the derived struct) has been resolved in `COWMacro.expansion(of:, providingMembersOf:, in:)`:

```swift
let storageTypeAndAssociatedMembers = factory.getStorageTypeDecl()
let storageTypeDecl = storageTypeAndAssociatedMembers.storageType
    
// resolving `storageName`
let storageName = structDecl.copyOnWriteStorageName ?? defaultStorageName
```

We can put the evaluation of storage before the line invoking `factory.getStorageTypeDecl()`. Then pass the evaluated `storageName` to `COWExpansionFactory.getStorageTypeDecl` and ultimately get the resolved `storageName` here. The core idea of this suggestion here is that we can make the global constant `defaultStorageName` to be `private` in `COWMacros.swift.`. I think this access control is more cohesive than the current one.

Reply:
Nice catch!